### DISCUSSION TITLE
React Player v3 plan

### DISCUSSION THREAD

**Discussion by luwes** [cookpete/react-player#1843] [Ideas]
Hi all, I wanted to share a rough plan what we had in mind for the next major version (v3) of React player.

There were several requests for adding custom player controls (UI) to React player and we think it would be really cool if React player could be made compatible with [Media Chrome](https://github.com/muxinc/media-chrome). Media Chrome will not be included by default, but the React player internal API will require some breaking changes so those layers can talk to each other.

Media Chrome supports players that conform to the HTMLMediaElement API and we build out many custom media elements over the years that do just that. All those media elements live since very recent in this monorepo https://github.com/muxinc/media-elements. They're all React compatible via the `/react` submodule import.

So in the spirit of compatibility and maintainability these elements will become the react-player "providers". The nice thing is that these API's will already be uniform coming into react-player. 

The downside is that some providers that are in react-player are not yet in media-elements.
These include Dailymotion, Facebook, Kaltura, MixCloud, SoundCloud, Streamable, Twitch and Vidyard.

The goal is to support many of those again if there is enough interest but the first v3 will not include these by default.
The ability to add custom players will always be supported so that is an escape hatch until they get added.

If you have any thoughts on this please let us know. We'd love to get feedback.

**Comment by heff** [comment]
Will this change the API of React Player or just the Provider API? As in Media Chrome can speak directly to the Providers without going through the React Player API?

**Reply by luwes** [reply]
yes, we can keep the user React API via props the same. potentially add some events or so but keep it backward compatible.

**Comment by AshesOfOwls** [comment]
Hey there,

I wanted to throw in my feedback that Twitch support would be very useful for the project I am working on which relies on it. V2 is working out great right now but I would love to migrate to V3. If you would like I can give more context on the project.

**Reply by luwes** [reply]
https://github.com/muxinc/media-elements/issues/127

**Comment by yvonnetangsu** [comment]
If we could get SoundCloud support for v3, that'd be awesome 🙏🏼

**Reply by luwes** [reply]
https://github.com/muxinc/media-elements/issues/131

**Comment by jy95** [comment]
Would be great if we have back the feature to support Youtube playlist urls https://github.com/cookpete/react-player/issues/1945 , like it was in V2

**Reply by jy95** [reply]
I created a PR to bring it back : https://github.com/muxinc/media-elements/pull/134

**Comment by vincentwinkel** [comment]
I don't know if it's the right place to say this, but please put back the support of Soundcloud and Mixcloud 🙏

**Reply by luwes** [reply]
https://github.com/muxinc/media-elements/issues/135

**Comment by javierdebug** [comment]
Hey, I have a question. Is the v3 going to be the only version at some point? If so, when do you think this will happen? Or do you plan to keep the support for v2? 
Thanks

**Reply by vincentwinkel** [reply]
I hope so because I just requested a fix for mixcloud and I need it merged asap for a website on production 😬

**Comment by binchik** [comment]
Hey! Thanks for the awesome lib! Is facebook video support planned?

**Comment by marcinheniborg** [comment]
Hi there! Thanks for the great lib! Is there a way to get the media title and the author from react-player?

**Comment by AshesOfOwls** [comment]
I'm back again to request Dailymotion to be added to media elements 🙂. Thanks again for the twitch and TikTok work.
